src/libraries/Microsoft.PowerFx.Core/Entities/QueryOptions/ExpandQueryOptions.cs

```diff
@@ -90,16 +90,28 @@ public bool AddSelect(string selectColumnName)
 
         public void AddRelatedColumns()
         {
-            if (!(ExpandInfo.ParentDataSource is IExternalCdsDataSource))
+            // This statement captures the properties of ExpandInfo and returns false iff all the captured values are
+            // non-null and of the correct type if specified
+            if (ExpandInfo is not {
+                    Identity: { } identity,
+                    ParentDataSource: IExternalCdsDataSource {
+                        Document: { GlobalScope: { } scope },
+                        DatasetName: { } datasetName } })
             {
+                Contracts.Assert(false, "The expand object lacks necessary dependencies.");
```

Comment:
Should this be a Check? Asserts in pfx only throw in pfx debug builds, not in host/canvas debug builds

Reply:
My thinking was that we just wouldn't throw.  We weren't throwing before if the parent data source is not iexternalcds data source.  We know this manifests as a runtime error.  Perhaps it's preferable to a crash?  What's the design philosophy here?